microservices/first/main.py

```diff
@@ -88,34 +98,50 @@ def update_deployment(deployment_id: int) -> tuple[Response, int]:
     """The `update` endpoint. Takes in the id of the SPEED build and the new
     results of the build. Used by leaders to update the web server on the
     build's progress in order to inform users via the `info` endpoint."""
+    app.logger.info("update endpoint invoked")
     data = request.json
     if not data:
+        app.logger.warning("internal endpoint `update` made bad request")
         return jsonify({"error": "Missing update data"}), 400
     db_manager: DBManager | None = getattr(g, "db_manager", None)
     if db_manager is None:
+        app.logger.error("unable to get DBManager")
         return jsonify({"error": "no database manager"}), 500
-    db_manager.update_deployment_fields(deployment_id, data)
-    return jsonify({"message": "Deployment updated successfully"}), 200
+    try:
+        db_manager.update_deployment_fields(deployment_id, data)
+        return jsonify({"message": "Deployment updated successfully"}), 200
+    except KeyError:
+        app.logger.warning("internal endpoint `update` made bad request")
+        return jsonify({"error": "missing necessary body data"}), 400
```

Comment:
In contrast this requires json data, however, this is really an internal endpoint as opposed to /start being external. So depends whether or not it is necessary for us to specify for internal endpoints.

Reply:
Technically, JSON data still arrives in the body, so I think this error message is still valid